patches/content-browser-renderer_host-render_frame_host_impl.cc.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/content/browser/renderer_host/render_frame_host_impl.cc b/content/browser/renderer_host/render_frame_host_impl.cc
+index 497033d3d5257daffe4aca2ebdd1fd71513db73f..32ab8f258aa8f69e33854f411b39d510b885ecf3 100644
+--- a/content/browser/renderer_host/render_frame_host_impl.cc
++++ b/content/browser/renderer_host/render_frame_host_impl.cc
+@@ -7840,6 +7840,7 @@ void RenderFrameHostImpl::BindSmsReceiverReceiver(
+ 
+ void RenderFrameHostImpl::BindRestrictedCookieManager(
+     mojo::PendingReceiver<network::mojom::RestrictedCookieManager> receiver) {
++  BRAVE_BINDRESTRICTEDCOOKIEMANAGER
```

Comment:
this patch is not necessary because there is already a way in ContentBrowserClient to create a special restricted cookie manager, but I think handling JS and network cookies separately is a mistake. I think we need to pass the 3p cookie info into the network service cookie manager so we can handle them all in the same way

Reply:
Our plan is to handle navigation cookies in a similar fashion, by implementing the abstract StoragePartition interface in `EphemeralStoragePartition`. Calls to the interface that do not access the `NetworkContext` will be routed to the default `StoragePartition`. Then we will use this `StoragePartition` when spinning up navigation in `NavigationRequest`. This work has bit-rotted a bit, but here is how this looks:

```diff
diff --git a/chromium_src/content/browser/renderer_host/navigation_request.cc b/chromium_src/content/browser/renderer_host/navigation_request.cc
index 635d7b75fd..5bf6d73bee 100644
--- a/chromium_src/content/browser/renderer_host/navigation_request.cc
+++ b/chromium_src/content/browser/renderer_host/navigation_request.cc
@@ -3,6 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "brave/content/browser/ephemeral_storage_partition.h"
 #include "content/browser/renderer_host/frame_tree.h"
 #include "content/browser/renderer_host/frame_tree_node.h"
 #include "url/gurl.h"
@@ -46,6 +47,18 @@ GURL GetTopDocumentGURL(content::FrameTreeNode* frame_tree_node) {
       GetTopDocumentGURL(frame_tree_node_), frame_tree_node_->IsMainFrame(), \
       common_params_->method, &common_params_->referrer);
 
+#define BRAVE_ONSTARTCHECKSCOMPLETE_REPLACESTORAGEPARTITION                  \
+  const GURL top_level_url = GetTopDocumentGURL(frame_tree_node_);           \
+  const url::Origin top_level_origin = url::Origin::Create(top_level_url);   \
+  const url::Origin target_origin = url::Origin::Create(common_params_->url);\
+  if (!navigating_frame_host->is_main_frame() &&                             \
+      !top_level_origin.IsSameOriginWith(target_origin)) {                   \
+    EphemeralStoragePartition* ephemeral_partition =                         \
+        browser_context->GetExistingEphemeralStoragePartition(top_level_url);\
+    if (ephemeral_partition)                                                 \
+      partition = ephemeral_partition;                                       \
+  }
+
 #include "../../../../../content/browser/renderer_host/navigation_request.cc"
```